client/src/templates/Introduction/components/block.test.tsx

```diff
@@ -0,0 +1,126 @@
+        fields: {} as Fields,
+        forumTopicId: 12345,
+        guideUrl: 'https://mockurl.com',
+        head: ['mockHead'],
+        hasEditableBoundaries: false,
+        helpCategory: 'mockHelpCategory',
+        id: 'mockId',
+        instructions: 'mockInstructions',
+        isComingSoon: false,
+        internal: {
+          content: 'mockContent',
+          contentDigest: 'mockContentDigest',
+          description: 'mockInternalDescription',
+          fieldOwners: ['mockOwner'],
+          ignoreType: null,
+          mediaType: 'mockMediaType',
+          owner: 'mockOwner',
+          type: 'mockType'
+        },
+        notes: 'mockNotes',
+        prerequisites: [] as PrerequisiteChallenge[],
+        removeComments: false,
+        isLocked: false,
+        isPrivate: false,
+        order: 1,
+        question: {} as Question,
+        assignments: ['mockAssignment'],
+        required: [],
+        solutions: {
+          ['indexhtml']: {} as FileKeyChallenge,
+          ['scriptjs']: {} as FileKeyChallenge,
+          ['stylescss']: {} as FileKeyChallenge,
+          ['indexjsx']: {} as FileKeyChallenge
+        },
+        sourceInstanceName: 'mockSourceInstanceName',
+        superOrder: 1,
+        superBlock: SuperBlocks.UpcomingPython,
+        tail: ['mockTail'],
+        template: 'mockTemplate',
+        tests: [] as Test[],
+        time: 'mockTime',
+        title: 'mockTitle',
+        translationPending: false,
+        url: 'https://mockurl.com',
+        usesMultifileEditor: false,
+        videoId: 'mockVideoId',
+        videoLocaleIds: {},
+        bilibiliIds: {} as BilibiliIds,
+        videoUrl: 'https://mockvideourl.com'
+      }
+    }
+  ],
+  completedChallengeIds: ['testchallengeIds'],
+  isExpanded: false,
+  t: jest.fn((key: string) => [key]),
+  toggleBlock: jest.fn()
+};
+
+function renderWithRedux(ui: JSX.Element) {
+  return render(<Provider store={createStore()}>{ui}</Provider>);
+}
+
+describe('<Block />', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('The "Help us translate" badge does not appear on any English blocks', () => {
+    renderWithRedux(
+      <Block {...defaultProps} superBlock={SuperBlocks.RespWebDesign} />
+    );
+    expect(
+      screen.queryByText(/misc.translation-pending/)
+    ).not.toBeInTheDocument();
+  });
+  it('The "Help us translate" badge does not appear on any i18n blocks when the superblock is audited', () => {
+    renderWithRedux(
+      <Block {...defaultProps} superBlock={SuperBlocks.RespWebDesign} />
+    );
+    expect(
+      screen.queryByText(/misc.translation-pending/)
+    ).not.toBeInTheDocument();
+  });
+
+  it('Help us translate badge does appear on i18n blocks when the superblock is not audited', () => {
+    renderWithRedux(
+      <Block {...defaultProps} superBlock={SuperBlocks.UpcomingPython} />
+    );
+    expect(screen.getByText(/misc.translation-pending/)).toBeInTheDocument();
```

Comment:
Am I missing something? How is this passing when the `env` seems to be only mocked for English?

Reply:
When the parameters language - 'english', superblock - 'UpcomingPython', showUpcomingChanges - 'false', and showNewCurriculum - 'true' are passed, the function createFlatSuperBlockMap will modify the Upcoming stage in the superBlockMap object, setting it to an empty array. This means that the UpcomingPython superblock will not be included in the flattened result array. Consequently, when these parameters are used in the isAudited function, the result will be false, indicating that the UpcomingPython superblock is not audited under these specific conditions.  

FYI just one thing I have to correct is I am passing flags values as boolean but in the code its expected to be as string 'true' or 'false'.

Reply:
It seems that the `env` mocked `showNewCurriculum` as `true` and `showUpcomingChanges` as `false`, so the `auditedSuperBlocks` in the `getAuditedSuperBlocks` function does not include `SuperBlocks.UpcomingPython`(`'upcoming-python'`), because in the `createFlatSuperBlockMap` function, `superBlockMap[SuperBlockStages.Upcoming]` became `[]` so `flatSuperBlockMap` does not include `SuperBlocks.UpcomingPython`.
Therefore, `isAuditedCert` will return `false` so `'misc.translation-pending'` will be included the `Block` component.

But the purpose of this test is that The "Help us translate" badge does appear on i18n blocks when the superblock is not audited. Should we loop through the all `SuperBlocks` with all `Languages` and test the `Block` component whether it includes `'misc.translation-pending'` or not?

Reply:
closing this conversation as its resolved and now we have a simpler approch for the test cases.